pilot/pkg/model/jwks_resolver.go

```diff
@@ -266,17 +265,35 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 // BuildLocalJwks builds local Jwks by fetching the Jwt Public Key from the URL passed if it is empty.
 func (r *JwksResolver) BuildLocalJwks(jwksURI, jwtIssuer, jwtPubKey string) *envoy_jwt.JwtProvider_LocalJwks {
 	if jwtPubKey == "" {
-		var err error
+		// var err error
 
 		// jwtKeyResolver should never be nil since the function is only called in Discovery Server request processing
 		// workflow, where the JWT key resolver should have already been initialized on server creation.
-		jwtPubKey, err = r.GetPublicKey(jwtIssuer, jwksURI)
-		if err != nil {
-			log.Errorf("Failed to fetch jwt public key from issuer %q, jwks uri %q: %s", jwtIssuer, jwksURI, err)
-			// This is a temporary workaround to reject a request with JWT token by using a fake jwks when istiod failed to fetch it.
-			// TODO(xulingqing): Find a better way to reject the request without using the fake jwks.
+		_, _ = r.GetPublicKey(jwtIssuer, jwksURI)
```

Comment:
Ignoring both return values on a method called "Get" seems very confusing. if its purpose is something else, maybe it should be renamed like "update cache" or something more appropriate?

Reply:
Planning the change this.